Okta.Core/OktaHttpClient.cs

```diff
@@ -175,6 +176,24 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpRequestType requestTy
                     {
                         httpClient.DefaultRequestHeaders.Remove("Authorization");
                     }
+                    if (bAddXForwardedHeader)
+                    {
+                        string ipAddress = null;
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            content = content.Trim();
+                            if (content.StartsWith("{") && content.EndsWith("}"))
+                            {
+                                var obj = JObject.Parse(content);
+                                ipAddress = (string)obj["context"]["ipAddress"];

```

Comment:
Deserializing our own content feels awkward. Ideally, we should refactor the `Post` method to accept an AuthContext, or some other entity that would contain this information. There may not be any easy way around this in the current architecture; thoughts @lboyette-okta or @jmelberg-okta?